doc/nrf/libraries/shell/shell_nfc.rst

```diff
@@ -0,0 +1,132 @@
+    used in an interrupt context.
+  * If library logs and the logger are enabled, then the logger should not be used in the synchronous mode.
+    Synchronous logs processing can introduce to big latency in NFC events processing which causing the NFC shell backend being unstable.
+    If you need logs from this library, configure logger to be in the deferred mode by setting the :kconfig:option:`CONFIG_LOG_MODE_DEFERRED` Kconfig option.
+
+Configuration
+*************
+
+Set the :kconfig:option:`CONFIG_SHELL_NFC` Kconfig option to enable the shell backend.
+
+The following configuration options are available for this module:
+
+   * :kconfig:option:`CONFIG_SHELL_NFC_INIT_PRIORITY` sets initialization priority for the NFC shell backend.
+   * :kconfig:option:`CONFIG_SHELL_NFC_BACKEND_TX_BUFFER` sets the NFC T4T APDU response buffer size.
+     It size might depends on your Reader/Writer maxiumum supported APDU size.
+   * :kconfig:option:`CONFIG_SHELL_NFC_BACKEND_TX_RING_BUFFER_SIZE` sets the shell Tx ring buffer size.
+     It is used to buffer data coming from shell interface for the NFC T4T transport, that can send data only on the data reception from the Reader/Writer.
+   * :kconfig:option:`CONFIG_SHELL_NFC_BACKEND_RX_RING_BUFFER_SIZE`- sets the shell Rx ring buffer size.
+     It handles shell incoming data from the NFC transport.
+
+See the Kconfig help for details.
+
+Usage
+*****
+
+You can set the :kconfig:option:`CONFIG_SHELL_NFC` Kconfig option to enable the NFC shell.
+The shell is initialized automatically on the system initialization.
+You can approach your Reader/Writer device that uses the raw ISO-DEP (ISO14443-4A) protocol to start the shell interface.
+Next, forward the received data to the terminal and send back data received from terminal through your Reader/Writer device.
+For example, it can be the keyboard data or the bulk data.
+
+.. note::
+   The NFC transport can be physically disconnected and any time. After disconnection all data sent by the shell will be discarded.
+   The incomplete commands sent by the Reader/Writer before transport lost are discarded also.
+   Every transport connection or re-connection starts a new shell session.
+
+Samples using the library
+*************************
+
+The following |NCS| samples use this library:
+
+* :ref:`nfc_shell`
+
+Limitations
+***********
+
+The NFC T4T ISO-DEP transport is relatively slow.
+The maximum bit rates is 106 kbits/s for NFC-A technology.
+Data is sent in blocks and Reader/Writer device needs to send data first or send dummy data to poll data from the Tag.
+This cause that, human visible latency occurs during printing characters on the terminal.
+It is not recommended also to set NFC shell as logger backend, due to transport speed.
+You can set the :kconfig:option:`CONFIG_SHELL_NFC_INIT_LOG_LEVEL_NONE` Kconfig option that, set the initial log level to none for the NFC shell.
```

Comment:
```suggestion
In the :kconfig:option:`CONFIG_SHELL_NFC_INIT_LOG_LEVEL_NONE` Kconfig option you can set the initial log level to ``none`` for the NFC shell.
```

Reply:
I used your suggestion but I am not sure in 100 % here